lib/elixir/test/elixir/calendar/holocene.exs

```diff
@@ -78,50 +79,81 @@ defmodule Calendar.Holocene do
     )
   end
 
+  defp zero_pad(val, count) when val >= 0 do
+    num = Integer.to_string(val)
+    :binary.copy("0", max(count - byte_size(num), 0)) <> num
+  end
+
+  defp zero_pad(val, count) do
+    "-" <> zero_pad(-val, count)
+  end
+
   @impl true
-  defdelegate time_from_day_fraction(day_fraction), to: Calendar.ISO
+  def parse_date(string) do
+    {year, month, day} =
+      string
+      |> String.split("-")
+      |> Enum.map(&String.to_integer/1)
+      |> List.to_tuple()
+
+    if valid_date?(year, month, day) do
+      {:ok, {year, month, day}}
+    else
+      {:error, :invalid_date}
+    end
+  end
 
   @impl true
-  defdelegate time_to_day_fraction(hour, minute, second, microsecond), to: Calendar.ISO
+  def valid_date?(year, month, day) when year >= 10_000 do
+    Calendar.ISO.valid_date?(year - 10_000, month, day)
```

Comment:
We can use `:calendar.valid_date`, which is not restricted to ISO dates. :)

Reply:
Done